### PR TITLE
Mobile 4115/improve sample apps

### PR DESCRIPTION
**Pull request by Mthomas3**

### CODE REVIEW

**Review by mathieu-ouvrard — CHANGES_REQUESTED** [review]
It's a huge PR, so I will sum up my comments here.

First remember that this repo is for our customers, it's public and hopefully many developers will see it (not the case for now, but we will try to communicate more on it).

Apart from a pure reorganisation, the goal of this task is to improve it as much as we can.
I would like to make it more visually compelling with images and striking layout.
I understand that you may not feel ready for such a change:
 - if you feel creative 👉 don't hesitate to add banners and logos and text layouts and Easter eggs and many things in the README and in the different apps, go wild !!
 - if not, no problem 👉 we will make another task ;-)

Now for more down to earth remarks:
 - there are still comment headers in some files
 - replicate the same features in every project: code duplication isn't a problem but we have to show how to integrate the SDK in the most explicit and easy way possible
 - you can rewrite some code if you think it's outdated but keep it on the same level feature level (you can even add some SDK API usage if you see some topics not covered by the current sample)
 - for consistency's sake, can you redo the Carthage project to name it similarly to the other ones
 - the SPM readme is quite strange. I know you just trim it but think of a customer not knowing CS and reading the README, think of what information he wants to get here
 - the SPM project is pointing at an exact SDK version, can you update to "up to next Major" ?
 - for the static project, it's usually a bad habit to upload binary files on git because git doesn't support binary diff (it recreate a full copy of the file each time). So I would advise against having the SDK In the repo, but more a README with a link to download the latest static release and instructions to copy it in the correct folder to be able to launch the app.
But it may be simpler as you did and not important to have the binary in the repo, I let you decide (you can read about git and binary files here for instance: https://vegeta2102.medium.com/why-you-should-not-push-binary-files-to-git-and-what-to-do-if-you-do-29f34692cef0)

**Review by mathieu-ouvrard — CHANGES_REQUESTED** [review]
Looking great !

- in all readme there is a typo on `Waht's Next`
- the link to the doc points to the sample section, the beginning of the doc would be better
- the static readme isn't like the others
- the CS logo in masking doesn't appear for me in the pod example
- the static app is different from the other ones

File: Static Framework Sample/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+### This iOS application will show you how to use Contentsquare SDK with Static framework dependency. <br />
+
+## Prerequisites
+
+Go to [Contentsquare github](https://github.com/ContentSquare/CS_iOS_SDK/releases), under Assets you should be able to find `ContentsquareModuleDynamicManually.xcframework.zip` file, please download it.
+
+Once you open the zip file on your computer, you should have two directories named `ContentsquareModule.xcframework` and `SwiftProtobuf.xcframework`.
+
+Copy both of them and paste it inside `ContentsquareModule` folder of the [current project](https://github.com/ContentSquare/iOS-sample-app/tree/mobile-4115/Improve-sample-apps/Static%20Framework%20Sample/ContentsquareModule).
```
Comment (mathieu-ouvrard) [inline Static Framework Sample/README.md:11]: I think git doesn't keep track of folders but only files.
So if someone does a first checkout of this repo, he won't have a `ContentsquareModule` folder so the instructions will quite misleading.
Maybe change the framework location at the root level ?

Comment (Mthomas3) [reply Static Framework Sample/README.md:11]: Indeed Git doesn't keep track of empty folders, but I added a .gitkeep inside the folder so when you clone it you still have it. If you think it's not a good idea and its better to put it at the root level, I will change it? :)

Comment (mathieu-ouvrard) [reply Static Framework Sample/README.md:11]: Well done, I didn't see it, good for me.

**Review by mathieu-ouvrard — COMMENTED** [review]
Can you update the SPM project to use the Rules: Version Up to Next Major 4.0.0 < 5.0.0

**Review by mathieu-ouvrard — APPROVED** [review]

**Comment by shravanteegala** [comment]
Hi @Mthomas3 ,
How i can identify specificScreen or View having masking or Unmasking. how i track it.

**Pull request merged**